packages/perseus/src/renderer.tsx

```diff
@@ -487,13 +487,18 @@ class Renderer extends React.Component<Props, State> {
         },
         props: Props,
     ) => any = (allWidgetInfo, props) => {
+        const {apiOptions, problemNum} = props;
         // @ts-expect-error - TS2345 - Argument of type '{ [key: string]: PerseusWidget; }' is not assignable to parameter of type 'Partial<Record<string, CategorizerWidget>>'.
-        return mapObject(allWidgetInfo, (widgetInfo) => {
+        const widgetsStartProps = mapObject(allWidgetInfo, (widgetInfo) => {
```

Comment:
I wonder if this is an opportunity to clean some old code. The `mapObject` helper seems confused about types so we have use expect error.
```suggestion
   type PerseusWidgetMap = {
            [key: string]: PerseusWidget;
        }
    _getAllWidgetsStartProps: (
        allWidgetInfo: PerseusWidgetMap,
        props: Props,
    ) => PerseusWidgetMap = (allWidgetInfo, props) => {
        const {apiOptions, problemNum} = props;
        const widgetsStartProps: PerseusWidgetMap = {};
        entries(allWidgetInfo).forEach([key, widgetInfo]) => {
            widgetsStartProps[key] = Widgets.getRendererPropsForWidgetInfo(
                widgetInfo,
                problemNum,
            );
        });
        apiOptions.getAllWidgetsStartProps?.(widgetsStartProps);
        return widgetsStartProps;
    };
```